pkg/operator/internalimageregistry/cleanup_controller.go

```diff
@@ -64,9 +64,46 @@ func (c *imagePullSecretCleanupController) sync(ctx context.Context, controllerC
 	if imageRegistryEnabled {
 		return nil
 	}
+
+	// once the image registry is disabled we want to wait unil the controller manager
+	// operand is deployed and stable before clean up the image pull secrets. we don't
+	// want to compete with the operand that may be acting on the same resources we are.
```

Comment:
There is still a chance for the controllers to be out of sync here, but I do think that this is an improvement.

Reply:
Can you please elaborate a little bit more on the `legacyImagePullSecretRollbackController` matter ? It does not seem to be in use in the `openshift-controller-manager` project (I could not find where it is started there).

I failed to see how the controller could be out of sync, can you give me more details ?